### ISSUE TITLE
 Convert the Field Controller PID tuning into the webapp

### ISSUE THREAD

**Issue by madhephaestus** [WPIRoboticsEngineering/RBE1001Lib#9]
Also, i was looking at brads web server code, iu think that the webapp could replace the java application. Thoughts?

**Comment by madhephaestus** [comment]
@acamilo can you take this one?

**Comment by madhephaestus** [comment]
@gcl8a said:

I realize it's recreating the filed controller, but a page on the web server, .../motors, could be useful for tuning the motors. I'm surprised how much difference they have, though, so no single defaults are going to work for everyone.

I could see fields for K and then a button to make the wheel spin once or something like that. They don't need to tune to perfection, but at least get rid of the jitters. This would save a lot of uploading test codes -- easily the single biggest time saver of the field controller.

**Comment by madhephaestus** [comment]
This feature would have a lot in common with a web based o-scope we discussed as well. The scope is something we would like to have for 2001, but would also be a useful feature in the 1001's, perhaps provide the kids with 2 esps and have oen set up as the scope, and the other as the robot.

**Comment by acamilo** [comment]
So, you want a page "motors" where you can enter in parameters. in the firmware, have a function that gets called when someone submits a form on that page. Also, some way of sending setpoints and having a graph of motor positions.

**Comment by madhephaestus** [comment]
Yeah, we can work together on the motor interface. Inside the Motor class is a static list of all of the motors, as well as a static field telling you how many there are. 

Look at the field controller for litterally what we want.

**Comment by madhephaestus** [comment]
Ope this file https://github.com/WPIRoboticsEngineering/2001_Field_Controller/blob/2001_Field/src/main/resources/fxml/MainScreen.fxml in SceneBuilder to see the wysiwyg of the UI

**Comment by gcl8a** [comment]
I don’t know what all goes into it, but I’d move for something “lightweight” here. In the end, for 1001 we’ll have two motors to control. So you could just have for each motor 3 fields (Kp, Ki, Kd) and a “commit” button that just calls Motor::setGains(). (You could probably get away with one set of gains — in fact that might be a better start.)

it’s trivial to do with the Serial Monitor, so if it’s a lot of work on the server side, we could just use a cable.

**Comment by madhephaestus** [comment]
Graph:
Setpoint
Position
Speed

With a 10 second sampling window

Set:
Setpoint
Gains (for all motors)

**Comment by acamilo** [comment]
So
SetSetpoint
setGains
